tests/models/test_mixins.py

```diff
@@ -52,52 +53,132 @@ def test_get_restrictions(self):
+
+        element.abstract = True
+        self.assertTrue(element.is_abstract)
 
-class ElementBaseTests(TestCase):
     def test_property_is_attribute(self):
-        obj = ElementBase.create()
-        self.assertFalse(obj.is_attribute)
+        element = ElementBase()
+        self.assertFalse(element.is_attribute)
 
-    def test_property_default_value(self):
-        obj = ElementBase.create()
-        self.assertIsNone(obj.default_value)
-
-        obj = el.Element.create(default="a")
-        self.assertEqual("a", obj.default_value)
-        self.assertFalse(obj.is_fixed)
-
-        obj.default = None
-        obj.fixed = 2
-        self.assertEqual(2, obj.default_value)
-        self.assertTrue(obj.is_fixed)
-
-    @mock.patch.object(ElementBase, "extends", new_callable=PropertyMock)
-    def test_extensions(self, mock_extends_property):
-        mock_extends_property.side_effect = [None, "", "foo  bar"]
-        obj = ElementBase.create()
-        self.assertEqual([], list(obj.extensions))
-        self.assertEqual([], list(obj.extensions))
-        self.assertEqual(["foo", "bar"], list(obj.extensions))
+    def test_property_is_fixed(self):
+        element = ElementBase()
+        self.assertFalse(element.is_fixed)
+
+        element.fixed = None
+        self.assertFalse(element.is_fixed)
+
+        element.fixed = "foo"
+        self.assertTrue(element.is_fixed)
+
+    def test_property_is_mixed(self):
+        element = ElementBase()
+        self.assertFalse(element.is_mixed)
+
+    def test_property_is_qualified(self):
+        element = ElementBase()
+        self.assertFalse(element.is_qualified)
+
+        element.form = None
+        self.assertFalse(element.is_qualified)
+
+        element.form = FormType.UNQUALIFIED
+        self.assertFalse(element.is_qualified)
+
+        element.form = FormType.QUALIFIED
+        self.assertTrue(element.is_qualified)
+
+        element = ElementBase()
+        element.form = FormType.UNQUALIFIED
+        element.ref = None
+        self.assertFalse(element.is_qualified)
+
+        element.ref = "foo"
+        self.assertTrue(element.is_qualified)
+
+    def test_property_prefix(self):
+        element = ElementBase()
+        self.assertIsNone(element.prefix)
+
+        element.ref = "foo"
+        self.assertIsNone(element.prefix)
+
+        element.ref = "foo:bar"
+        self.assertEqual("foo", element.prefix)
+
+    def test_raw_namespace(self):
+        element = ElementBase()
+        self.assertIsNone(element.raw_namespace)
+
+        element.target_namespace = "tns"
+        self.assertEqual("tns", element.raw_namespace)
+
+    def test_raw_type(self):
+        element = ElementBase()
+        self.assertIsNone(element.raw_namespace)
+
+        element.type = "xs:int"
+        self.assertEqual("xs:int", element.raw_type)
+
+    def test_real_name(self):
+        element = ElementBase()
+
+        with self.assertRaises(NotImplementedError):
+            element.real_name
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Statement seems to have no effect](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=5096260)